app/models/flag.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+class Flag < ActiveRecord::Base
+  scope :enabled, -> { where(enabled: true) }
+
+  before_save :set_position
+  after_save :reset_flag_settings!
+  after_destroy :reset_flag_settings!
+
+  def used?
+    PostAction.exists?(post_action_type_id: self.id) ||
+      ReviewableScore.exists?(reviewable_score_type: self.id)
+  end
+
+  def self.reset_flag_settings!
+    PostActionType.reload_types
```

Comment:
I think it would be worth doing a writeup about how the `PostActionType` and `ReviewableScore` flag types differ. All of this intertwining with `PostActionType` makes me nervous, but maybe we can't do much about it for historical reasons. I just want us to be super clear about how the flagging system fits together especially when it comes to custom flags defined by admins.